guava/src/com/google/common/base/Equivalence.java

```diff
@@ -398,4 +409,29 @@ private Object readResolve() {
 
     private static final long serialVersionUID = 1;
   }
+
+  static final class BiPredicateEquivalence<V> extends Equivalence<V> implements Serializable {
+
+    private final transient BiPredicate<? super V, ? super V> predicate;
+
+    BiPredicateEquivalence(BiPredicate<? super V, ? super V> predicate) {
+      this.predicate = checkNotNull(predicate);
+    }
+
+    @Override
+    protected boolean doEquivalent(V a, V b) {
+      return this.predicate.test(a, b);
+    }
+
+    @Override
+    protected int doHash(V v) {
+      return v.hashCode();
+    }
+
+    public static <V> BiPredicateEquivalence<V> of(BiPredicate<? super V, ? super V> predicate) {
+      return new BiPredicateEquivalence<>(predicate);
+    }
```

Comment:
This method doesn't need to be `public`, since the class isn't. (Maybe this method isn't necessary all all, since it merely delegates to the constructor?)